server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Email.scala

```diff
@@ -91,8 +91,13 @@ object Email {
     defaultMessageBuilder.setDecodeMonitor(DecodeMonitor.SILENT)
     val inputStream = firstMessage.getFullContent.getInputStream
     val resultMessage = Try(defaultMessageBuilder.parseMessage(inputStream))
-    inputStream.close()
-    resultMessage
+    resultMessage.fold(e => {
+      inputStream.close()
```

Comment:
even if i think your solution looks nicer, i think you should wrap the close in a Try as it can throw an IO exception. Which would not be catched this way.